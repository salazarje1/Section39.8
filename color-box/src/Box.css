.Parent{
    display: inline-block;
    position: relative;
}

.Box{
    display: inline-block;
    margin: 1rem;
}

.Box-button {
    display: inline; 
    position: absolute;
    left: 20px;
    top: 20px;
    border: none;
    font-size: 1em;
}