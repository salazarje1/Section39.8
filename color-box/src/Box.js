import React from 'react'; 

import './Box.css'; 


const Box = ({removeBox, id, color, width=3, height=3}) => {
    return (
        <div className='Parent'>
            <div 
                className='Box'
                style={{
                    backgroundColor: color,
                    width: `${width * 20}px`,
                    height: `${height * 20}px`
                }}
            ></div>
            <button style={{backgroundColor: color}} className='Box-button' onClick={() => removeBox(id)}>X</button>
        </div>
        
    )
}


export default Box; 