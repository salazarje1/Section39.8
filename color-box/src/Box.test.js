import { render } from '@testing-library/react'; 
import Box from './Box';

test('box renders without crashing', () => {
    render(<Box />); 
})

test('it matches snapshot', () => {
    const { asFragment } = render(<Box />)

    expect(asFragment()).toMatchSnapshot(); 
})