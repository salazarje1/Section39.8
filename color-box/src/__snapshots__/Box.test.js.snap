// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Parent"
  >
    <div
      class="Box"
      style="width: 60px; height: 60px;"
    />
    <button
      class="Box-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
